### PR TITLE
BlockHeadHair doesn't make players bald.

### PR DESCRIPTION
**Pull request by nachomeep**
Instead, it replaces their hairstyle with short hair [unless their hair was already short enough to not have any clipping issues]. Also replaces a lot of 'BLOCKHAIR' tags with 'BLOCKHEADHAIR', because chaplains should have beards visible when their hoods are on.

Tested. It works. No clipping issues found. Point out any other hair that should be allowed and any that should be disallowed, and any items that need tag updates regarding BLOCKHAIR and BLOCKHEADHAIR.

Mostly ported from - https://github.com/Baystation12/Baystation12/pull/14239

### CODE REVIEW

**Comment by nachomeep** [comment]
Forgot a changelog. Will be commiting in a bit.

File: code/modules/mob/new_player/sprite_accessories.dm [hunk 1]
```diff
@@ -547,12 +562,10 @@
 /datum/sprite_accessory/facial_hair
 
 	icon = 'icons/mob/Human_face.dmi'
-	gender = MALE // barf (unless you're a dorf, dorfs dig chix /w beards :P)
```
Comment (Neerti) [inline code/modules/mob/new_player/sprite_accessories.dm]: Not use if removing this actually does anything, but this appears to be an undocumented change.

Comment (nachomeep) [inline code/modules/mob/new_player/sprite_accessories.dm]: Oh, sorry. I'll change that back- that was for testing purposes.

**Comment by Anewbe** [comment]
Mind going back through this and not touching any of the gender related flags?

**Comment by nachomeep** [comment]
Not at all.

**Comment by Anewbe** [comment]
Alright, code looks fine now. Have you tested this?

**Comment by nachomeep** [comment]
I have. Haven't found any clipping issues or anything thus far.

**Pull request merged**